word_guess_lv.rb

```diff
@@ -0,0 +1,312 @@
+# The user will have 4 trials to enter a level. We did this to have an end of the program
+# in case the user keeps entering the wrong level input.
+  def selection_level
+
+    levels = %w[low medium high]
+    puts "Please enter a level: low, medium or high?"
+    level = nil
+    until level == "low" || level == "medium" || level == "high"
+      level = gets.chomp.downcase
+      if level == "low"
+        @word_array = @words_low
+      elsif level == "medium"
+        @word_array = @words_medium
+      elsif level == "high"
+        @word_array = @words_high
+      elsif level != "low" && level != "hard" && level != "medium"
+        puts "Invalid level. Please enter again"
+        level = gets.chomp.downcase
+      end
+
+      attempt = 0
+      while !levels.include?(level)
+        if attempt < 3
+          attempt += 1
+          puts "Please enter a valid level. You have #{(-attempt + 3) + 1} more attempts.\n"
+          level = gets.chomp
+        else
+          puts "Too many attempts..start the program again"
+          exit
+        end
+      end
+    end
+
+  end
+
+# This method hanbles inappropriate user input.
+# This method validates that the input of letter from the user is a letter and not any other character.
+# We did this to don't count inputs as wrong guesses if the user inputs numbers, symbols etc.
+  def validate_input_letter(input)
+    until !input.match(/[^a-z]/)
+      puts "\nPlease return a valid letters from A to Z.\n"
+      input = gets.chomp.downcase
+    end
+  end
+
+# In this method, the user inputs a letter. The method can validate input is a letter,
+# and also can tell if the letter has been already used.
+  def letters_input
+    @guesses_counter = 0
+    while @guesses_counter < 5
+      puts "Guess a letter!"
+      puts "Word: #{@letters_correct.join("")}"
+      letter = gets.chomp
+
+      validate_input_letter(letter)
+
+# This conditional checks in case the user enters the whole word
+      if letter.length > 1
+        if letter == @word
+          you_win
+          puts "------> WOW...you guessed the word!".bold.blink
+          exit
+        elsif letter != @word
+          puts "That's not the correct word but you can try letters!"
+        end
+      end
+
+#This loop helps the user to don't be penalized for guessing the same letter more than once.
+      while @letters_guessed.include? letter do
+        puts "You have already tried this letter, try another one: "
+        letter = gets.chomp
+
+        validate_input_letter(letter)
+      end
+
+      @letters_guessed << letter
+
+      if @word.include? letter
+        @word.split(//).each_index do |i|
+          if @word[i] == letter
+            @letters_correct[i] = letter
+          end
+        end
+        ascii_image
+      else
+        @guesses_counter += 1
+        puts "Keep trying!!".colorize(:orange).blink
+        ascii_image
+      end
+      if @word.split(//) == @letters_correct
+        you_win
+        exit
```

Comment:
There are several places in your code where you `exit`. This is fine for this assignment, but it means your program is limited to only playing the game once. An interesting exercise is to think about how you would have to change it for the user to be able to play again.